.grit/patterns/CheckZero.md

```diff
@@ -0,0 +1,88 @@
+---
+title: Require Checking for Zero Addresses
+---
+
+# Require Checking for Zero Addresses
+
+Ensure that all functions that take an address as a parameter check that it is not zero. Note: this only works with inline functions.
+
+```grit
+language sol
+
+// Marks the body of a function as "checked"
+pattern CheckedBody($address) = bubble($address) contains or {
+    // Directly checks the require
+    `require($address != address(0), $_)`
+    `require($address != address(0))`
+    `require(address(0) != $address, $_)`
+    `require(address(0) != $address)`
+
+    // Or look for a function call
+    sol_call_expression(function=`$id`, children=contains $address) where {
+        $program <: contains bubble($id) sol_function_definition(name=$id, children=$child2) where {
+            // Inspect the body of the transitive function recursively
+            $child2 <: CheckedBody($address)
```

Comment:
Should discuss with cos, but iirc current implementation of scopes in Marzano is not conducive to recursive pattern definitions.